workflows/src/main/kotlin/com/r3/corda/lib/ci/RequestKeyFlow.kt

```diff
@@ -92,11 +87,19 @@ class ProvideKeyFlow(private val otherSession: FlowSession) : FlowLogic<Unit>()
     override fun call() {
         val request = otherSession.receive<SendRequestForKeyMapping>().unwrap { it }
         when (request) {
-            is RequestKeyForUUID -> otherSession.send(createSignedOwnershipClaimFromUUID(serviceHub, request.challengeResponseParam, request.externalId))
-            is RequestForKnownKey -> otherSession.send(createSignedOwnershipClaimFromKnownKey(serviceHub, request.challengeResponseParam, request.knownKey))
+            is RequestKeyForUUID -> {
```

Comment:
No need to change anything but a code style thing: you could return the key from the when block then call `registerKeyToParty` only once at the end of the function.